### ISSUE TITLE
Generic `useNextItem` hook

### ISSUE THREAD

**Issue by joshxg** [openmsupply/openmsupply-client#1062]
There are a lot of `useNextItem` hooks for each of the current documents which handles the `ok & next` state for add/edit modals.

There is probably a nice solution to make this into a single generic hook

**Comment by mark-prins** [comment]
Agree, it would be good to have a generic hook and consolidate the logic
